airflow/providers/google/cloud/hooks/bigquery.py

```diff
@@ -2077,6 +2082,9 @@ def run_query(
         labels = labels or self.labels
         schema_update_options = list(schema_update_options or [])
 
+        if priority:
+            self.priority = priority
+
```

Comment:
I think you should not update the class instance attribute:
```python
# if you create a hook with default priority BATCH
hook = BigQueryHook(..., priority="BATCH")
# the you run a query with priority INTERACTIVE
hook.run_query(..., priority="INTERACTIVE")
# finally, if you run another query without priority argument, it should use the one
# provided on the instance creation (BATCH), but it is not the case
hook.run_query(....)  # self.priority is set to INTERACTIVE
```

Reply:
I'm OK working off `BigQueryHook()` but was wondering if the preferred way to submit arbitrary sql is to leverage `SQLExecuteQueryOperator()` under commons.

I'm pretty new to airflow and wasn't sure when I should start with operator vs starting with hook. It didn't seem like there was an obvious way of passing in a modified hook into `SQLExecuteQueryOperator()`

If I use `BigQueryInsertJobOperator()` (the bigquery specific way of doing `SQLExecuteQueryOperator()`) then I would need to create a custom operator based on `BigQueryInsertJobOperator()` that overwrites `_submit_job()` since that calls `BigQueryHook()` and operator class also doesn't have `priority` as a class attribute.

Reply:
Why not just add a priority parameter to BigQueryInsertJobOperator as well? That way, you can instantiate the hook with it? Please default it in such a way where it is backwards compatible.

Reply:
In `BigQueryInsertJobOperator` priority is set by `configuration`. This field is similar to the `api_resource_configs` field within `BigQueryHook`